docs/reference/pages/model_recipes.rst

```diff
@@ -163,39 +165,98 @@ Using an example from the Wagtail demo site, here's what the tag model and the r
 
 Wagtail's admin provides a nice interface for inputting tags into your content, with typeahead tag completion and friendly tag icons.
 
-Now that we have the many-to-many tag relationship in place, we can fit in a way to render both sides of the relation. Here's more of the Wagtail demo site ``models.py``, where the index model for ``BlogPage`` is extended with logic for filtering the index by tag:
+We can now make use of the many-to-many tag relationship in our views and templates. For example, we can set up the blog's index page to accept a ``?tag=...`` query parameter to filter the ``BlogPage`` listing by tag:
 
 .. code-block:: python
 
     from django.shortcuts import render
 
     class BlogIndexPage(Page):
         ...
-        def serve(self, request):
-            # Get blogs
-            blogs = BlogPage.objects.child_of(self).live()
+        def get_context(self, request):
+            context = super().get_context(request)
+
+            # Get blog entries
+            blog_entries = BlogPage.objects.child_of(self).live()
 
             # Filter by tag
             tag = request.GET.get('tag')
             if tag:
-                blogs = blogs.filter(tags__name=tag)
+                blog_entries = blog_entries.filter(tags__name=tag)
 
+            context['blog_entries'] = blogs
```

Comment:
`context['blog_entries'] = blog_entries`

Also, shouldn't `render()` use `context` rather than the custom dict?